### ISSUE TITLE
[Cross-platform Dotnet NuGet Sign] Creating a new test certificate fails with the message “The specified item is no longer valid. It may have been deleted from the keychain” on MacOS Sequoia 15.0 

### ISSUE THREAD

**Issue by CiciLi1** [NuGet/Home#13825]
### NuGet Product Used

dotnet.exe

### Product Version

.NET SDK Version: 8.0.403-servicing.24463.3  

### Worked before?

This is a regression on MacOS Sequoia 15.0

### Impact

It's more difficult to complete my work

#### Notes:    

1. The repro rate is 100%.   
2. This issue reproes on MacOS: Sequoia 15.0, it doesn’t repro on MacOS Sonoma 14.6.1. 

 

#### Steps:  

1. [Patch dotnet SDK](https://microsoft.sharepoint.com/teams/NuGet/_layouts/15/Doc.aspx?sourcedoc=%7B8a3c6685-1661-47f4-8807-b765ca62aea1%7D&action=edit&wd=target%28Package%20Signing%20Manual%20Test.one%7C0344a15b-01b1-4828-af4c-63fe53687f8a%2FPatch%20dotnet%20SDK%7Cfd0e3c41-e7a8-4de8-b83a-1a09fdb56271%2F%29&wdorigin=703): Patch dotnet SDK. 
2. Go to the patchedSDK folder and try to create a new test certificate with the command below:  
`./dotnet run --project ./Entropy/MakeTestCert/MakeTestCert.csproj --framework net8.0 --password password `

     

#### Expected:  

Creating the new test certificate is successful. 

#### Actual:        

Creating a new test certificate fails with the message “The specified item is no longer valid. It may have been deleted from the keychain” as below:
![Unbenannt](https://github.com/user-attachments/assets/83ccd406-22d8-4a65-a89c-d6681b5a7ecb)

**Comment by nkolev92** [comment]
@kartheekp-ms, @dtivel, can you help here? 

During Client Triage, there was a discussion that there were some APIs deprecated in the newest MacOS that may be the root cause of this issue.

@CiciLi1 Does 8.0.402-servicing.24419.7 patched with Dev\6.12.0.87 work on MacOS: Sequoia 15.0

**Comment by CiciLi1** [comment]
> @kartheekp-ms, @dtivel, can you help here?
> 
> During Client Triage, there was a discussion that there were some APIs deprecated in the newest MacOS that may be the root cause of this issue.
> 
> @CiciLi1 Does 8.0.402-servicing.24419.7 patched with Dev\6.12.0.87 work on MacOS: Sequoia 15.0

Hi @nkolev92, this issue is also repro on MacOS: Sequoia 15.0 with Dev\6.12.0.87 patch 8.0.402-servicing.24419.7.

**Comment by nkolev92** [comment]
Sounds like it's only a MacOS: Sequoia 15.0 problem then.

**Comment by dtivel** [comment]
This is similar to https://github.com/dotnet/runtime/issues/106775 in callstack and error message but not in scenario.  In that scenario, users are trying to register dev certs, so suggested workarounds don't apply here.

I reproduced this on my macOS running 15.0.1 and vanilla 8.0.303, which was installed globally before the macOS upgrade and was not patched with latest NuGet.  I then installed 8.0.403 (like the issue suggested) and the problem went away.  Even when I forced 8.0.303 through a global.json file, I still couldn't reproduce the issue again.

Try installing (or reinstalling) a global (not test) .NET SDK to see if that fixes this issue.  Try using the global dotnet CLI to run MakeTestCert.

This seems like a post-upgrade Sequoia issue that eventually goes away.
